repose-aggregator/services/service-client/impl/src/test/scala/org/openrepose/services/serviceclient/akka/impl/AkkaServiceClientImplTestToo.scala

```diff
@@ -0,0 +1,117 @@
+import org.junit.runner.RunWith
+import org.mockito.Matchers._
+import org.mockito.AdditionalMatchers.or
+import org.mockito.Mockito.when
+import org.openrepose.services.httpclient.{HttpClientResponse, HttpClientService}
+import org.scalatest.junit.JUnitRunner
+import org.scalatest.mock.MockitoSugar
+import org.scalatest.{BeforeAndAfter, FunSpec, Matchers}
+
+import scala.collection.JavaConversions._
+
+@RunWith(classOf[JUnitRunner])
+class AkkaServiceClientImplTestToo extends FunSpec with BeforeAndAfter with Matchers with MockitoSugar with LazyLogging {
+  val HEADER_SLEEP = "Origin-Sleep"
+  val RESPONSE_BODY = "RESPONSE BODY"
+  val LIST_APPENDER_REF = "List0"
+  val httpClientService = mock[HttpClientService[AnyRef, HttpClientResponse]]
+  val httpClientResponse = mock[HttpClientResponse]
+  val httpClient = new DefaultHttpClient
+  val params = httpClient.getParams
+  var request: HttpGet = _
+  val originServer = new Server(0)
+  val hashKey = "hashKey"
+  var app: ListAppender = _
+  var httpResponse: HttpResponse = _
+  var uri: String = _
+
+  before {
+    val ctx = LogManager.getContext(false).asInstanceOf[LoggerContext]
+    val cfg = ctx.getConfiguration
+    app = cfg.getAppender(LIST_APPENDER_REF).asInstanceOf[ListAppender]
+    when(httpClientService.getMaxConnections(or(anyString(), isNull.asInstanceOf[String]))).thenReturn(20)
+    when(httpClientService.getClient(or(anyString(), isNull.asInstanceOf[String]))).thenReturn(httpClientResponse)
+    when(httpClientResponse.getHttpClient).thenReturn(httpClient)
+    originServer.setHandler(new AbstractHandler() {
+      override def handle(s: String, request: Request, httpServletRequest: HttpServletRequest, httpServletResponse: HttpServletResponse): Unit = {
+        logger.trace("Starting handle...")
+        val timeout = httpServletRequest.getHeader(HEADER_SLEEP)
+        logger.trace(s"timeout=$timeout")
+        if (timeout != null) {
+          logger.trace("Sleeping...")
+          Thread.sleep(timeout.toLong)
+          logger.trace("Continuing...")
+        }
+        httpServletResponse.setContentType(request.getHeader(HttpHeaders.ACCEPT))
+        httpServletResponse.setStatus(HttpServletResponse.SC_OK)
+        logger.trace("httpServletResponse.getStatus=" + httpServletResponse.getStatus)
+        httpServletResponse.getOutputStream.println(RESPONSE_BODY)
+        logger.trace(s"httpServletResponse.getBody=$RESPONSE_BODY")
+        request.setHandled(true)
+        logger.trace("Finished handle.")
+      }
+    })
+    originServer.start()
+    val port = originServer.getConnectors.toList.get(0).asInstanceOf[ServerConnector].getLocalPort
+    uri = s"http://localhost:$port"
+    request = new HttpGet(uri)
+    request.setHeader(HttpHeaders.ACCEPT, MediaType.APPLICATION_XML)
+  }
+
+  after {
+    originServer.stop()
+  }
+
+  val timeouts = List(2000/*, 30000, 45000, 55000, 90000*/)
+  timeouts.foreach { timeout =>
```

Comment:
Sure did. I just didn't want to leave these in the actual test suite due to it adding over 3 mins to the test.